packages/components/src/components/Progress/ProgressItem.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useEffect, useState } from "react";
+import { LayoutAnimation, View } from "react-native";
+import { useStyles } from "../../hooks/useStyles";
+import { useToken } from "../../hooks/useToken";
+import { EDSStyleSheet } from "../../styling";
+import { Button } from "../Button";
+import { Typography } from "../Typography";
+import { ProgressExpandButton } from "./ProgressExpandButton";
+import { ProgressItemStatus } from "./ProgressItemStatus";
+import { ProgressTaskItem } from "./ProgressTaskItem";
+import { ProgressStatus, ProgressTask, ProgressTaskErrorDetails } from "./types";
+import { computeTaskStatus } from "./progressUtils";
+
+type ProgressItemPropsOptions =
+    | {
+          /**
+           * An array of tasks to be tracked by the progress item. Each task represents a unit of work whose progress or status can be monitored.
+           * Specifying `tasks` will automatically determine the overall status of the progress item based on the individual statuses of these tasks.
+           */
+          tasks: ProgressTask[];
+          /**
+           * The `status` prop should not be used when `tasks` is provided. The overall status is computed based on the progress of the individual tasks.
+           */
+          status?: never;
+      }
+    | {
+          /**
+           * When `tasks` is not provided, the `status` prop can be used to manually set the overall status of the progress item.
+           * This is useful for simpler use cases where there is a single task or the progress does not need to be broken down into individual tasks.
+           */
+          tasks?: never;
+          /**
+           * Manually set the overall status of the progress item. Valid statuses; 'notStarted', 'inProgress', 'success', or 'error'.
+           */
+          status: ProgressStatus;
+      };
+
+export type ProgressItemProps = {
+    /**
+     * Optional title for the progress item
+     */
+    title?: string;
```

Comment:
Maybe nitpicking, but should we require the `title` prop instead? No title or description would mean an empty progress item, right?
Is having blank items needed / wanted? Maybe having it be required is good for ensuring users of the component stick to design standards